pkg/app/run.go

```diff
@@ -345,8 +346,11 @@ func configureKingpin(data *global.Data) *kingpin.Application {
 //
 // Finally, we check if there is a profile override in place (e.g. set via the
 // --profile flag or using the `profile` field in the fastly.toml manifest).
-func processToken(cmds []argparser.Command, data *global.Data) (token string, tokenSource lookup.Source, err error) {
-	token, tokenSource = data.Token()
+func processToken(commandName string, cmds []argparser.Command, data *global.Data) (token string, tokenSource lookup.Source, err error) {
+	// if the command is 'profile token', then the token can only
```

Comment:
This is necessary because if a token is found on the command line or in the environment, no validity-checking will be performed... and then the `profile token` command will output the possibly-expired token from the profile.

Verified in my testing.